File: src/main/java/com/campus/activity/entity/Comment.java
package com.campus.activity.entity;

import lombok.Data;
import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

@Data
public class Comment {

    private Long id;

    @NotNull(message = "活动ID不能为空")
    private Long activityId;

    @NotNull(message = "用户ID不能为空")
    private Long userId;

    @NotBlank(message = "评论内容不能为空")
    @Length(min = 1, max = 1000, message = "评论内容必须在1-1000字符之间")
    private String content;

    private Long replyToId;

    private LocalDateTime createdAt;

    private LocalDateTime deletedAt;
}

File: web/src/types/comment.ts
/** 评论信息 */
export interface Comment {
  id: number;
  activityId: number;
  userId: number;
  username: string;
  content: string;
  replyToId: number | null;
  createdAt: string;
}

/** 评论创建请求 */
export interface CommentRequest {
  content: string;
  replyToId?: number;
}

/** 评论响应 */
export interface CommentResponse {
  id: number;
  activityId: number;
  userId: number;
  username: string;
  content: string;
  replyToId: number | null;
  replyToUsername: string | null;
  createdAt: string;
  replyCount: number;
  replies: CommentResponse[];
}
